File: bin/mudband/android/Mudband/app/src/main/java/band/mud/android/ui/UiDashboardUserTosAgreement.kt
/*-
 * Copyright (c) 2024 Weongyo Jeong <weongyo@gmail.com>
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
 * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
 * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
 * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
 * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
 * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 * SUCH DAMAGE.
 */

package band.mud.android.ui

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.SideEffect
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalUriHandler
import androidx.compose.ui.unit.dp
import androidx.navigation.NavHostController
import band.mud.android.ui.model.MudbandAppViewModel

@Composable
fun UiDashboardUserTosAgreement(
    viewModel: MudbandAppViewModel,
    navController: NavHostController,
    modifier: Modifier = Modifier
) {
    val uriHandler = LocalUriHandler.current

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(16.dp)
    ) {
        LazyColumn(
            modifier = Modifier
                .weight(1f)
                .fillMaxWidth()
        ) {
            item {
                Text(
                    text = "User Agreement",
                    style = MaterialTheme.typography.titleMedium,
                    modifier = Modifier.padding(vertical = 8.dp)
                )
            }
            item {
                Text(
                    text = "The following information is collected while mud.band app is running to perform P2P (Peer To Peer) connection:",
                    style = MaterialTheme.typography.bodyMedium,
                    modifier = Modifier.padding(vertical = 8.dp)
                )
            }
            item {
                Text(
                    text = "Public IP",
                    modifier = Modifier.padding(vertical = 4.dp)
                )
                Text(
                    text = "Private / Local IP",
                    modifier = Modifier.padding(vertical = 4.dp)
                )
            }
            item {
                Text(
                    text = "No other information except above are logged at mud.band server side. " +
                            "By clicking \"I agree\", you agree to our terms of service and privacy policy.",
                    style = MaterialTheme.typography.bodyMedium,
                    modifier = Modifier.padding(vertical = 8.dp)
                )
            }
        }

        Row {
            Button(
                onClick = {
                    viewModel.setUserTosAgreement(true)
                }
            ) {
                Text(text = "I agree")
            }
            Spacer(modifier = Modifier.weight(0.1f))
            Button(
                onClick = {
                    uriHandler.openUri("https://www.mud.band/policy/tos")
                }
            ) {
                Text(text = "View ToS / Privacy Policy")
            }
        }
    }

    SideEffect {
        viewModel.setTopAppBarTitle("Mud.band")
    }
}

File: bin/mudband/macos/Mudband/Mudband/ui/UiDashboardUserTosAgreement.swift
/*-
 * Copyright (c) 2025 Weongyo Jeong <weongyo@gmail.com>
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
 * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
 * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
 * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
 * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
 * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 * SUCH DAMAGE.
 */

import Alamofire
import Foundation
import SwiftyJSON
import SwiftUI

struct UiDashboardUserTosAgreement: View {
    @EnvironmentObject private var mAppModel: AppModel
    @Environment(\.openURL) var openURL

    var body: some View {
        VStack {
            Text("User Agreement")
                .padding()
                .font(.largeTitle)
            Text("The following information is collected while mud.band app is running to perform P2P (Peer To Peer) connection:")
            List {
                Section {
                    Text("Public IP")
                    Text("Private / Local IP")
                    Text("NAT type")
                }
            }
            Text("No other information except for the above is logged on the mud.band server, and the data will not be shared with any third parties." + " " + "By clicking the \"I agree\" button, you agree to our Terms of Service and Privacy Policy.")
                .padding()
            HStack {
                Button {
                    mAppModel.set_user_tos_agreement(agreed: true)
                } label: {
                    Text("I agree")
                }
                Button {
                    if let url = URL(string: "https://www.mud.band/policy/tos") {
                        openURL(url)
                    }
                } label: {
                    Text("View ToS / Privacy Policy")
                }
            }
            .padding()
        }
    }
}

struct UiDashboardUserTosAgreementPreviews: PreviewProvider {
    static var previews: some View {
        UiDashboardUserTosAgreement()
    }
}
